internal/config/nsmgr/nsmgr.go

```diff
@@ -0,0 +1,157 @@
+package nsmgr
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	nspkg "github.com/containernetworking/plugins/pkg/ns"
+	"github.com/containers/storage/pkg/idtools"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+// NamespaceManager manages the server's namespaces.
+// Specifically, it is an interface for how the server is creating namespaces.
+// and can be requested to create namespaces for a pod.
+type NamespaceManager interface {
```

Comment:
Personally I find the practice of using an interface that only has a single implementation excessive. I might be overlooking something, but what are the chances cri-o will have one more implementation of `NamespaceManager`? If chances are low, we should not be use an interface here, as this is an abstraction that comes with a price tag.

Reply:
that's a great point. there originally were two, but then we dropped `manage_ns_lifecycle`, and I don't see there being a new one any time soon